packages/runtime/src/widgets/ToggleButton.tsx

```diff
@@ -45,47 +45,72 @@ export type ToggleButtonProps = {
 } & EnsembleWidgetProps<ToggleButtonPropsStyles & EnsembleWidgetStyles>;
 
 export const ToggleButton: React.FC<ToggleButtonProps> = ({
-  id,
+  id: propId,
   value: givenValue,
   items,
   onChange,
   styles,
 }) => {
   const [value, setValue] = useState(givenValue);
-  const { values, rootRef } = useRegisterBindings({ value, styles }, id, {
-    setValue,
-  });
+  const { values, rootRef, id } = useRegisterBindings(
+    { value, styles },
+    propId,
+    {
+      setValue,
+    },
+  );
+
+  // onchange action handler
   const action = useEnsembleAction(onChange);
   const onChangeCallback = useCallback(
-    (_event: React.MouseEvent<HTMLElement>, newValue: string) => {
+    (newValue: string) => {
       if (!action) {
         return;
       }
```

Comment:
So the issue was we don't update the value if there is no action?

Reply:
Yes